assemblies/assembly-planner-java.adoc

```diff
@@ -0,0 +1,768 @@
+                            .collect(Collectors.joining(" | "))
+                    + " |");
+            LOGGER.info("|            | "
+                    + cellList.stream().map(cellLessonList -> String.format("%-10s",
+                            cellLessonList.stream().map(Lesson::getStudentGroup).collect(Collectors.joining(", "))))
+                            .collect(Collectors.joining(" | "))
+                    + " |");
+            LOGGER.info("|" + "------------|".repeat(roomList.size() + 1));
+        }
+        List<Lesson> unassignedLessons = lessonList.stream()
+                .filter(lesson -> lesson.getTimeslot() == null || lesson.getRoom() == null)
+                .collect(Collectors.toList());
+        if (!unassignedLessons.isEmpty()) {
+            LOGGER.info("");
+            LOGGER.info("Unassigned lessons");
+            for (Lesson lesson : unassignedLessons) {
+                LOGGER.info("  " + lesson.getSubject() + " - " + lesson.getTeacher() + " - " + lesson.getStudentGroup());
+            }
+        }
+    }
+
+}
+----
+
+The `main()` method first creates the `SolverFactory`:
+
+[source,java]
+----
+SolverFactory<TimeTable> solverFactory = SolverFactory.create(new SolverConfig()
+        .withSolutionClass(TimeTable.class)
+        .withEntityClasses(Lesson.class)
+        .withConstraintProviderClass(TimeTableConstraintProvider.class)
+        // The solver runs only for 5 seconds on this small dataset.
+        // It's recommended to run for at least 5 minutes ("5m") otherwise.
+        .withTerminationSpentLimit(Duration.ofSeconds(5)));
+----
+
+This registers the `@PlanningSolution` class, the `@PlanningEntity` classes,
+and the `ConstraintProvider` class, all of which you created earlier.
+
+Without a termination setting or a `terminationEarly()` event, the solver runs forever.
+To avoid that, the solver limits the solving time to five seconds.
+
+After five seconds, the `main()` method loads the problem, solves it, and prints the solution:
+
+[source,java]
+----
+        // Load the problem
+        TimeTable problem = generateDemoData();
+
+        // Solve the problem
+        Solver<TimeTable> solver = solverFactory.buildSolver();
+        TimeTable solution = solver.solve(problem);
+
+        // Visualize the solution
+        printTimetable(solution);
+----
+
+The `solve()` method doesn't return instantly. It runs for five seconds before returning the best solution.
+
+OptaPlanner returns _the best solution_ found in the available termination time.
+Due to
+//<<doesPlannerFindTheOptimalSolution>>,
+the nature of NP-hard problems,
+the best solution might not be optimal, especially for larger datasets.
+Increase the termination time to potentially find a better solution.
+
+The `generateDemoData()` method generates the school time table problem to solve.
+
+The `printTimetable()` method pretty prints the timetable to the console,
```

Comment:
```suggestion
The `printTimetable()` method prettyprints the timetable to the console,
```
Prettyprint is one word (I think). See https://en.wikipedia.org/wiki/Prettyprint 

Should we even use "prettyprint" at all? Seems informal.